File: SchoolWebApp-Client/src/app/cbe/assessments/models/pci.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class PCI extends ResourceModel<PCI> {
    public name?: string;
    public description?: string;
    public rank?: number;
    public subStrandId?: number;
    public subStrand?: any;

    // Used by add/edit form for cascading filters
    public academicYearId?: number;
    public curriculumId?: number;
    public learningLevelId?: number;
    public subjectId?: number;
    public strandId?: number;

    constructor(model?: Partial<PCI>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/PCI.cs
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class PCI: SettingsBase
    {
        public int SubStrandId { get; set; }
        public SubStrand? SubStrand { get; set; }
    }
}
